src/Analyzers/Core/Analyzers/AbstractCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -59,10 +61,11 @@ internal abstract class AbstractCodeStyleDiagnosticAnalyzer : DiagnosticAnalyzer
         protected AbstractCodeStyleDiagnosticAnalyzer(ImmutableArray<DiagnosticDescriptor> supportedDiagnostics)
         {
             SupportedDiagnostics = supportedDiagnostics;
-        }
 
-        protected DiagnosticDescriptor CreateUnnecessaryDescriptor()
-            => CreateUnnecessaryDescriptor(DescriptorId);
+            Descriptor = SupportedDiagnostics[0];
```

Comment:
Ideally the subtype should never access `Descriptor` if it was constructed through this constructor with multiple `supportedDiagnostics`. Instead of this change, I'd suggest converting `Descriptor` to a property, whose getter throws an invalid operation exception if the underlying descriptor is null.

Reply:
I looked into this but it doesn't really do anything to reduce the chance of misuse. Ideally, we eliminate the `DiagnosticDescriptor` properties in the base class and require derived types construct them directly (possibly by calling static methods of the base class).